src/hub/ShareClassManager.sol

```diff
@@ -325,69 +325,43 @@ contract ShareClassManager is Auth, IShareClassManager {
 
         UserOrder storage userOrder = depositRequest[scId_][depositAssetId][investor];
         require(userOrder.pending > 0, NoOrderFound());
+
         require(userOrder.lastUpdate <= issueEpochId[scId_][depositAssetId], IssuanceRequired());
         canClaimAgain = userOrder.lastUpdate < issueEpochId[scId_][depositAssetId];
 
         EpochInvestAmounts storage epochAmounts = epochInvestAmounts[scId_][depositAssetId][userOrder.lastUpdate];
 
-        if (epochAmounts.approvedAssetAmount == 0) {
-            emit ClaimDeposit(
-                poolId,
-                scId_,
-                userOrder.lastUpdate,
-                investor,
-                depositAssetId,
-                depositAssetAmount,
-                userOrder.pending,
-                payoutShareAmount,
-                epochAmounts.issuedAt
-            );
-            userOrder.lastUpdate += 1;
-            return (payoutShareAmount, depositAssetAmount, cancelledAssetAmount, canClaimAgain);
-        }
-
-        // Skip epoch if user cannot claim
-        depositAssetAmount = _fulfillRatio(epochAmounts).mulUint128(userOrder.pending);
+        depositAssetAmount = epochAmounts.approvedAssetAmount == 0
+            ? 0
+            : userOrder.pending.mulDiv(epochAmounts.approvedAssetAmount, epochAmounts.pendingAssetAmount).toUint128();
```

Comment:
@mustermeiszer Unfortunately your newly added `_fullfillRatio` calculations are less precise. During the initial work on the SCM, I spent some time for that and eventually concluded that `pendingUser * approved * pendingTotal` is more precise.

Reply:
Isn't there a way to always make it `uint256` based so that we don't lose precision? ^^

Reply:
Sorry for rolling it back, and gread you found the issue.